pkgs/by-name/au/autosubst2/package.nix

```
@@ -0,0 +1,49 @@
+{
+  lib,
+  haskellPackages,
+  fetchFromGitHub,
+}:
+
+haskellPackages.mkDerivation {
+  pname = "autosubst2";
+  version = "unstable-2022-07-04";
+
+  src = fetchFromGitHub {
+    owner = "uds-psl";
+    repo = "autosubst2";
+    rev = "8a71e1dc4dea81f13a9572ea302064eb374566c6";
+    hash = "sha256-3n87NBi3NbuDb44/oEnzjNk/TAHUwATJYuaw70k/tpk=";
+  };
+  patches = [ ./imports.patch ];
+
+  isLibrary = true;
+  isExecutable = true;
+  libraryHaskellDepends = with haskellPackages; [
+    base
+    parsec
+    containers
+    mtl
+    array
+    wl-pprint
+    graphviz
+    text
+  ];
+  executableHaskellDepends = with haskellPackages; [
+    base
+    optparse-applicative
+    wl-pprint
+    directory
+  ];
+  testHaskellDepends = with haskellPackages; [
+    base
+    QuickCheck
+    containers
+  ];
+  doCheck = false;
+
+  homepage = "https://github.com/uds-psl/autosubst2";
+  description = "Tool for generating de Bruijn boilerplate Coq code to handle substitutions in languages with binders";
+  maintainers = with lib.maintainers; [ chen ];
+  license = lib.licenses.bsd3;
+  mainProgram = "as2-exe";
```

These go inside `meta` and mean nothing as top-level attributes

```suggestion
meta = {
  homepage = "https://github.com/uds-psl/autosubst2";
  description = "Tool for generating de Bruijn boilerplate Coq code to handle substitutions in languages with binders";
  maintainers = with lib.maintainers; [ chen ];
  license = lib.licenses.bsd3;
  mainProgram = "as2-exe";
};
```

---

After changing this I got a build error:

```
➜  nixpkgs git:(init-autosubst2) nix-build -A autosubst2
error:
       … while calling a functor (an attribute set with a '__functor' attribute)
         at /home/chen/Development/nixpkgs/lib/customisation.nix:264:13:
          263|     in if missingArgs == {}
          264|        then makeOverridable f allArgs
             |             ^
          265|        # This needs to be an abort so it can't be caught with `builtins.tryEval`,

       … while evaluating a branch condition
         at /home/chen/Development/nixpkgs/lib/customisation.nix:148:7:
          147|     in
          148|       if isAttrs result then
             |       ^
          149|         result // {

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: function 'anonymous lambda' called without required argument 'license'
       at /home/chen/Development/nixpkgs/pkgs/development/haskell-modules/generic-builder.nix:32:1:
           31|
           32| { pname
             | ^
           33| , dontStrip ? outputsJS
```

I think the doc for the Meta attributes of `haskellPackages.mkDerivation` is saying that the meta info will be transparently set in `meta` of the resulting derivation by the builder: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/haskell.section.md#meta-attributes-haskell-derivation-meta

---

Ewwwwww... why does it do that!? Anyway, disregard then :+1: